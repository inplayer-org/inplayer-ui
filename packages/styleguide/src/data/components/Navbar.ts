import { ComponentData } from "../types";

const usage = `
  <Navbar 
    logo="https://inplayer.com/wp-content/themes/darkwin/pics/brand/android-icon-192x192.png">
    <Navbar.MenuItem icon={<h1>M1</h1>} active>
        Menu item 1
    </Navbar.MenuItem>
    <Navbar.MenuItem icon={<h1>M2</h1>}>Menu item 2</Navbar.MenuItem>
  </Navbar>
`;

const demoCode = `<Navbar logo="https://inplayer.com/wp-content/themes/darkwin/pics/brand/android-icon-192x192.png">
    <Navbar.MenuItem icon={<h5>M1</h5>} active>
    Menu item 1
    </Navbar.MenuItem>
    <Navbar.MenuItem
    icon={<h5>M2</h5>}
    onClick={() => console.log('Clicked menu item')}
    >
    Menu item 2
    </Navbar.MenuItem>
    <a
    style={{
        textDecoration: 'none',
    }}
    href="http://www.example.com"
    >
    <Navbar.MenuItem icon={<h5>EX</h5>}>External item</Navbar.MenuItem>
    </a>
</Navbar>
`;

const Navbar: ComponentData = {
  description: `The Navbar is a horizontal navigation bar component. It takes an image string or element as the logo icon at the top. Menu items are rendered as children of the Navbar using the Navbar.MenuItem component. The MenuItem takes an icon as a prop (which can be a string, which is added as a className to the icon, for FontAwesome CSS classes for example, or an actual React Node), and an active prop, which indicates whether the selected menu item is active. onClick is an optional prop which lets you add custom click logic to the MenuItem. The MenuItem can in an a or Link from react-router element as well, just make sure to clear the text-decoration.`,
  path: 'src/components/NavBar/index.js',
  usage,
  examples: [
    {
      title: 'Demo',
      code: demoCode,
    }
  ],
  propsAndMethods: [
    { name: 'children', type: 'Node', default: 'Required' },
    { name: 'logo', type: 'union', default: 'Required' },
    { name: 'className', type: 'string', description: 'A className can be passed down for further styling or extending with CSS-in-JS' },
    { name: 'children', type: 'Node', default: 'Required' },
  ],
}

export default Navbar;