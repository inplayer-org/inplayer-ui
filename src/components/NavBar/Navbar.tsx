import React, { useState } from 'react';

// Components
import Arrow from './Arrow';
import { NavbarContainer } from './styled';
import MenuItem from './MenuItem';
import MenuWithLogo from './MenuWithLogo';

type Props = {
  children: any;
  /**
   * Navbar logo
   */
  logo?: string;
};

const Navbar = ({ children, logo }: Props) => {
  const [open, setOpen] = useState(false);

  const toggleOpen = () => setOpen(!open);

  return (
    <NavbarContainer open={open} onClick={toggleOpen}>
      <MenuWithLogo logo={logo}>{children}</MenuWithLogo>
      <Arrow open={open} section="navbar" />
    </NavbarContainer>
  );
};

Navbar.MenuItem = MenuItem;

export default Navbar;
