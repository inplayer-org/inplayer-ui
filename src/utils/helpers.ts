import moment from 'moment';

export const getMonthOptions = (): Array<any> =>
  moment.months().map((label, value) => ({ value, displayName: label }));

export const getYearOptions = () => {
  const yearOptions: Array<any> = [];
  for (let i = moment().year() - 100; i <= moment().year(); i += 1) {
    yearOptions.push({ value: i.toString(), displayName: i.toString() });
  }

  return yearOptions;
};
