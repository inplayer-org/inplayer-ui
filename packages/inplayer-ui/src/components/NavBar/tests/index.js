import { generateSnapshots } from 'packages/inplayer-ui/src/components/NavBar/tests/helpers';
import Navbar from '../index';

generateSnapshots(Navbar);
