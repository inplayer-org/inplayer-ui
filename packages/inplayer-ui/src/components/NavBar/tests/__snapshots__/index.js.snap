// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
.c4 {
  width: 18px;
  margin: 0 1rem 1rem 1rem;
  text-align: center;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c4:hover i {
  border-color: #fff;
}

.c5 {
  display: inline-block;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
  cursor: pointer;
  padding: 0.225rem;
  border: solid rgba(255,255,255,0.4);
  border-width: 1px 0 0 1px;
  -webkit-transform: rotate(135deg);
  -ms-transform: rotate(135deg);
  transform: rotate(135deg);
}

.c0 {
  background-color: #253a66;
  width: 50px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: width 0.3s ease;
  transition: width 0.3s ease;
  cursor: pointer;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  overflow: hidden;
}

.c2 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  height: 34px;
  margin: 0.5rem 0;
}

.c3 {
  width: 100%;
  text-align: left;
  margin: 3.125rem 0 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
  onClick={[Function]}
  open={false}
  style={Object {}}
>
  <span
    className="c1"
  >
    <img
      className="c2"
    />
    <nav
      className="c3"
    />
  </span>
  <div
    className="c4"
  >
    <i
      className="c5"
      open={false}
    />
  </div>
</div>
`;
